app/controllers/menus_controller.rb

```diff
@@ -0,0 +1,54 @@
+class MenusController < ApplicationController
+
+  def index
+    @menus            = Menu.all
+    @page_content     = "page-content"
+    @header           = "header"
+    @footer           = "footer"
+  end
+
+  def show
+    @menu             = Menu.find(params[:menu_id])
+    @page_content     = "page-content"
+    @header           = "header"
+    @footer           = "footer"
+    @menus            = Menu.all
+  end
+
+  def preview
+    @menus            = Menu.all
+    @menu             = Menu.find(params[:menu_id])
+    @page_content     = "preview page-content"
+    @header           = "header-preview"
+    @footer           = "footer-preview"
+  end
+
+  def add
+    @page_content     = "page-content"
+    @menu             = Menu.new
+    @menu.name        = params[:menu_name]
+    if @menu.save
+      redirect_to all_menus_path
+    else
+      @menus          = Menu.all
+      @page_content     = "page-content"
+      @header           = "header"
+      @footer           = "footer"
+      render "menus/index"
+    end
+  end
+
+  def delete_menu
+    @menu = Menu.find(params[:menu_id])
+    @menu.destroy
+    redirect_to all_menus_path
+  end
+
+  def edit_name
```

Comment:
Typically actions are best if they are a little more generic, nameing this `edit_name` corners you a bit to imply that this only edits the name, but if you later decide that another column can be edited, it would require you to change the method name. something like `edit` or `update` would be more conventional.